File: AgregacionComposicion/Ejercicio7/Estudiante.py
class Estudiante:
    def __init__(self, nombre, carrera, semestre):
        self.__nombre = nombre
        self.__carrera = carrera
        self.__semestre = semestre
    
    # Getters y setters
    def get_nombre(self): return self.__nombre
    def set_nombre(self, nombre): self.__nombre = nombre
    def get_carrera(self): return self.__carrera
    def set_carrera(self, carrera): self.__carrera = carrera
    def get_semestre(self): return self.__semestre
    def set_semestre(self, semestre): self.__semestre = semestre
    
    def mostrar_info(self):
        print(f"Estudiante: {self.__nombre} | Carrera: {self.__carrera} | Semestre: {self.__semestre}")

class Universidad:
    def __init__(self, nombre):
        self.__nombre = nombre
        self.__estudiantes = []
    
    # Getters y setters
    def get_nombre(self): return self.__nombre
    def set_nombre(self, nombre): self.__nombre = nombre
    def get_estudiantes(self): return self.__estudiantes
    
    def agregar_estudiante(self, estudiante):
        if isinstance(estudiante, Estudiante):
            self.__estudiantes.append(estudiante)
    
    def mostrar_universidad(self):
        print(f"\nUniversidad: {self.__nombre}")
        print("Estudiantes inscritos:")
        for estudiante in self.__estudiantes:
            estudiante.mostrar_info()

# Ejemplo de uso
if __name__ == "__main__":
    # Estudiantes independientes
    est1 = Estudiante("Ana Torres", "Ing. Sistemas", 4)
    est2 = Estudiante("Carlos Ruiz", "Medicina", 2)
    
    # Universidad
    uni = Universidad("UNMSM")
    uni.agregar_estudiante(est1)
    uni.agregar_estudiante(est2)
    
    # Mostrar información
    uni.mostrar_universidad()
    
    # Estudiantes existen independientemente
    print("\nEstudiantes por separado:")
    est1.mostrar_info()
    est2.mostrar_info()

File: AgregacionComposicion/Ejercicio7/Estudiante.java
import java.util.ArrayList;
import java.util.List;

class Estudiante {
    private String nombre;
    private String carrera;
    private int semestre;
    
    public Estudiante(String nombre, String carrera, int semestre) {
        this.nombre = nombre;
        this.carrera = carrera;
        this.semestre = semestre;
    }
    
    // Getters y setters
    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }
    public String getCarrera() { return carrera; }
    public void setCarrera(String carrera) { this.carrera = carrera; }
    public int getSemestre() { return semestre; }
    public void setSemestre(int semestre) { this.semestre = semestre; }
    
    public void mostrarInfo() {
        System.out.printf("Estudiante: %s | Carrera: %s | Semestre: %d%n", 
                         nombre, carrera, semestre);
    }
}

class Universidad {
    private String nombre;
    private List<Estudiante> estudiantes;
    
    public Universidad(String nombre) {
        this.nombre = nombre;
        this.estudiantes = new ArrayList<>();
    }
    
    // Getters y setters
    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }
    public List<Estudiante> getEstudiantes() { return estudiantes; }
    
    public void agregarEstudiante(Estudiante estudiante) {
        if (estudiante != null) {
            estudiantes.add(estudiante);
        }
    }
    
    public void mostrarUniversidad() {
        System.out.printf("%nUniversidad: %s%n", nombre);
        System.out.println("Estudiantes inscritos:");
        for (Estudiante est : estudiantes) {
            est.mostrarInfo();
        }
    }
}

public class Main {
    public static void main(String[] args) {
        // Estudiantes independientes
        Estudiante est1 = new Estudiante("Luisa Méndez", "Derecho", 5);
        Estudiante est2 = new Estudiante("Jorge Castro", "Arquitectura", 3);
        
        // Universidad
        Universidad uni = new Universidad("UNI");
        uni.agregarEstudiante(est1);
        uni.agregarEstudiante(est2);
        
        // Mostrar información
        uni.mostrarUniversidad();
        
        // Estudiantes existen independientemente
        System.out.println("\nEstudiantes por separado:");
        est1.mostrarInfo();
        est2.mostrarInfo();
    }
}